siliconcompiler/remote/client.py

```diff
@@ -283,27 +284,35 @@ def remote_process(chip, steplist):
     Dispatch the Chip to a remote server for processing.
     '''
     _load_remote_config(chip)
+    should_resume = chip.get('option', 'resume')
+    remote_resume = should_resume and chip.get('record', 'remoteid')
 
     # Pre-process: Run an starting nodes locally, and upload the
     # in-progress build directory to the remote server.
     # Data is encrypted if user / key were specified.
     # run remote process
-    if chip.get('arg', 'step'):
+    if should_resume:
+        chip.unset('arg', 'step')
+        chip.unset('arg', 'index')
+    elif chip.get('arg', 'step'):
         chip.error('Cannot pass "-step" parameter into remote flow.', fatal=True)
     cur_steplist = chip.get('option', 'steplist')
     pre_remote_steplist = {
         'steplist': cur_steplist,
         'set': chip.schema._is_set(chip.schema._search('option', 'steplist')),
     }
-    _remote_preprocess(chip, steplist)
+    # Only run the pre-process step if the job doesn't already have a remote ID.
+    if not remote_resume:
+        # If we are resuming a local job remotely, set the steplist accordingly.
+        if should_resume:
+            steplist = cur_steplist if cur_steplist else chip.list_steps()
+        _remote_preprocess(chip, steplist)
 
     # Run the job on the remote server, and wait for it to finish.
     # Set logger to indicate remote run
     chip._init_logger(step='remote', index='0', in_run=True)
     _remote_run(chip)
 
-    # Delete the job's data from the server.
-    delete_job(chip)
```

Comment:
since we want this one quickly, could you make a separate PR with just this so we can get this in quickly.

Reply:
Sure.

Reply:
https://github.com/siliconcompiler/siliconcompiler/pull/1936